src/main/drivers/bus_spi_hal.c

```diff
@@ -91,128 +128,105 @@ void spiInitDevice(SPIDevice device, bool leadingEdge)
     spi->hspi.Init.FifoThreshold = SPI_FIFO_THRESHOLD_01DATA;
     spi->hspi.Init.MasterKeepIOState = SPI_MASTER_KEEP_IO_STATE_ENABLE;  /* Recommanded setting to avoid glitches */
 #endif
-
-    if (spi->leadingEdge) {
-        spi->hspi.Init.CLKPolarity = SPI_POLARITY_LOW;
-        spi->hspi.Init.CLKPhase = SPI_PHASE_1EDGE;
-    } else {
-        spi->hspi.Init.CLKPolarity = SPI_POLARITY_HIGH;
-        spi->hspi.Init.CLKPhase = SPI_PHASE_2EDGE;
-    }
+    spi->hspi.Init.CLKPolarity = SPI_POLARITY_LOW;
+    spi->hspi.Init.CLKPhase = SPI_PHASE_1EDGE;
 
     // Init SPI hardware
     HAL_SPI_Init(&spi->hspi);
 }
 
-// return uint8_t value or -1 when failure
-uint8_t spiTransferByte(SPI_TypeDef *instance, uint8_t out)
-{
-    uint8_t in;
-
-    spiTransfer(instance, &out, &in, 1);
-    return in;
-}
-
-/**
- * Return true if the bus is currently in the middle of a transmission.
- */
-bool spiIsBusBusy(SPI_TypeDef *instance)
-{
-    SPIDevice device = spiDeviceByInstance(instance);
-    if (spiDevice[device].hspi.State == HAL_SPI_STATE_BUSY) {
-        return true;
-    } else {
-        return false;
-    }
-}
-
-bool spiTransfer(SPI_TypeDef *instance, const uint8_t *out, uint8_t *in, int len)
+static bool spiPrivReadWriteBufPolled(SPI_TypeDef *instance, const uint8_t *txData, uint8_t *rxData, int len)
 {
     SPIDevice device = spiDeviceByInstance(instance);
     HAL_StatusTypeDef status;
 
-    if (!in) {
+    if (!rxData) {
         // Tx only
-        status = HAL_SPI_Transmit(&spiDevice[device].hspi, out, len, SPI_TIMEOUT_SYS_TICKS);
-    } else if (!out) {
+        status = HAL_SPI_Transmit(&spiDevice[device].hspi, txData, len, SPI_TIMEOUT_SYS_TICKS);
+    } else if(!txData) {
         // Rx only
-        status = HAL_SPI_Receive(&spiDevice[device].hspi, in, len, SPI_TIMEOUT_SYS_TICKS);
+        status = HAL_SPI_Receive(&spiDevice[device].hspi, rxData, len, SPI_TIMEOUT_SYS_TICKS);
     } else {
         // Tx and Rx
-        status = HAL_SPI_TransmitReceive(&spiDevice[device].hspi, out, in, len, SPI_TIMEOUT_SYS_TICKS);
+        status = HAL_SPI_TransmitReceive(&spiDevice[device].hspi, txData, rxData, len, SPI_TIMEOUT_SYS_TICKS);
     }
 
-    if (status != HAL_OK) {
-        spiTimeoutUserCallback(instance);
-    }
-
-    return true;
+    return (status == HAL_OK);
 }
 
-// Position of Prescaler bits are different from MCU to MCU
-
-static uint32_t baudRatePrescaler[8] = {
-    SPI_BAUDRATEPRESCALER_2,
-    SPI_BAUDRATEPRESCALER_4,
-    SPI_BAUDRATEPRESCALER_8,
-    SPI_BAUDRATEPRESCALER_16,
-    SPI_BAUDRATEPRESCALER_32,
-    SPI_BAUDRATEPRESCALER_64,
-    SPI_BAUDRATEPRESCALER_128,
-    SPI_BAUDRATEPRESCALER_256,
-};
-
-void spiSetDivisor(SPI_TypeDef *instance, uint16_t divisor)
+void spiPrivInitStream(const extDevice_t *dev, bool preInit)
```

Comment:
Private, as in this isn't part of the SPI API for use by drivers, but is only use internally by the SPI bus driver. As such it's defined in `bus_spi_impl.h` rather than `bus_spi.h` along with `spiPrivStartDMA` and `spiPrivStopDMA`. I realise that `spiResetStream` and `spiResetDescriptors` should be treated the same so I've renamed them to match.

Reply:
`Internal` is commonly used in Betaflight for internal functions.